kube/src/api/dynamic.rs

```diff
@@ -226,7 +232,7 @@ impl GroupVersionKind {
     /// Creates `GroupVersionKind` from group, version and kind.
     /// For `core` resources, group should be empty.
     /// `api_version` will be created based on group and version
-    pub fn from_dynamic_gvk(group: &str, version: &str, kind: &str) -> Self {
+    pub fn from_dynamic_gvk(group: &str, version: &str, kind: &str, plural: &str) -> Self {
```

Comment:
feel like 4 string parameters to a ctor for GVK is awkward. The 3 before was OK because it was the 3 in the name of the struct. Now I kind of think we should have a builder for GVK, and have it pluralize if plural is missed out.

Reply:
have changed GVK quite a bit in #474. But should be easier to add a mutator and the extra field with the way it's set up there.